sympy/functions/elementary/exponential.py

```diff
@@ -1234,30 +1234,54 @@ def _eval_is_algebraic(self):
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
         if len(self.args) == 1:
             arg = self.args[0]
-            arg0 = arg.subs(x, 0).cancel()
-            if not arg0.is_zero:
-                return self.func(arg0)
-            return arg.as_leading_term(x)
+            _, e = arg.as_coeff_exponent(x)
+            if e.is_nonnegative:
+                arg0 = arg.subs(x, 0).cancel()
+                if not arg0.is_zero:
+                    return self.func(arg0)
+                return arg.as_leading_term(x)
 
     def _eval_nseries(self, x, n, logx, cdir=0):
         if len(self.args) == 1:
             from sympy.functions.elementary.integers import ceiling
             from sympy.series.order import Order
             arg = self.args[0].nseries(x, n=n, logx=logx)
             lt = arg.as_leading_term(x, logx=logx)
-            lte = 1
+            lte = S.One
             if lt.is_Pow:
                 lte = lt.exp
-            if ceiling(n/lte) >= 1:
-                s = Add(*[(-S.One)**(k - 1)*Integer(k)**(k - 2)/
-                          factorial(k - 1)*arg**k for k in range(1, ceiling(n/lte))])
-                s = expand_multinomial(s)
-            else:
-                s = S.Zero
-
-            return s + Order(x**n, x)
+            if lte.is_positive:
+                if (ceiling(n/lte) - 1).is_nonnegative is not False:
+                    s = Add(*[(-S.One)**(k - 1)*Integer(k)**(k - 2)/
```

Comment:
Testing `is_nonnegative is not False` means that we go down this branch when `is_nonnegative is None`.

It is possible to use inequalities here but it needs to be like:
```python
if (ceiling(n / lte) >= 1) is S.true:
```

Reply:
See: https://docs.sympy.org/latest/guides/booleans.html

Reply:
Since it is necessary to evaluate `range(1, ceiling(n/lte))`, `ceiling(n/lte)` must be a positive integer.

```python
x = symbols('x', positive=True)
if (ceiling(x) >= 1) is S.true:
    range(1, ceiling(x)) # error
```

Therefore, we need to check if it is a positive integer as follows:
```python
c = ceiling(n/lte)
if c.is_Integer and c >= 1:
    ...
```